corehq/apps/geospatial/static/geospatial/js/gps_capture.js

```diff
@@ -159,14 +184,35 @@ hqDefine("geospatial/js/gps_capture",[
         return self;
     };
 
+    function updateSelectedItemLonLat(lon, lat) {
+        selectedDataListItem.setCoordinates(lat, lon);
+    }
+
+    function onMarkerDragEnd() {
+        let lngLat = coordinateCaptureMarker.getLngLat();
+        updateSelectedItemLonLat(lngLat.lng, lngLat.lat);
+    }
+
+    function setMarkerAtLngLat(lon, lat, centerView) {
+        if (coordinateCaptureMarker) { coordinateCaptureMarker.remove(); }
+
+        coordinateCaptureMarker = new mapboxgl.Marker({
+            color: selectedMarkerColor,
+            draggable: true,
+        });
+        coordinateCaptureMarker.on('dragend', onMarkerDragEnd);
+        coordinateCaptureMarker.setLngLat([lon, lat]);
+        coordinateCaptureMarker.addTo(map);
+    }
+
     var initMap = function () {
         'use strict';
 
         mapboxgl.accessToken = initialPageData.get('mapbox_access_token');  // eslint-disable-line no-undef
 
         let centerCoordinates = [2.43333330, 9.750]; // should be domain specific
 
-        const map = new mapboxgl.Map({  // eslint-disable-line no-undef
+        map = new mapboxgl.Map({  // eslint-disable-line no-undef
```

Comment:
Can you share why const was removed here?

Reply:
I declated `map` as a constant at the top of the file, because I need access to the `map` variable [here](https://github.com/dimagi/commcare-hq/blob/351fafbf8a0c1d950c1e28d9cfcc3401233509cd/corehq/apps/geospatial/static/geospatial/js/gps_capture.js#L34).

Reply:
I see it (const) was added back in a later commit.